Lib/test/test_codecs.py

```diff
@@ -3440,5 +3440,22 @@ def search_function(encoding):
         self.assertEqual(NOT_FOUND, codecs.lookup('a\xe9\u20ac-8'))
 
 
+class EncodingNormalizationTest(unittest.TestCase):
+
+    def test_bpo39337(self):
+        """
+        bpo-39337: similar to _Py_normalize_encoding(),
+        encodings.normalize_encoding() should ignore non-ASCII letters.
+        """
+        import encodings
+
+        out = encodings.normalize_encoding('utf\xE9\u20AC\U0010ffff-8')
+        self.assertEqual(out, 'utf_8')
+        out = encodings.normalize_encoding('utf_8')
+        self.assertEqual(out, 'utf_8')
+        out = encodings.normalize_encoding('utf   8')
+        self.assertEqual(out, 'utf_8')
```

Comment:
You may add a test with dots, since "." is special in the function.

Please add also a test with upper-case letters, to test that the function does not convert to lower case.

Reply:
Thanks, done.